### DISCUSSION TITLE
Is it possible to create multiple themes in one React App?

### DISCUSSION THREAD

**Discussion by mikalai-repin** [keycloakify/keycloakify#347] [General]
Hi!
I need to create multiple themes based on a single application. Is it possible to do so?

If I try to generate something new in build_keycloak/src/main/reources/themes, then on localhost:8080 I see only the last created theme available. 

keycloakify-starter used.

**Comment by garronej** [comment]
Hi @mikalai-repin,  
Thanks for reaching out, it depend of what you are talking about, if you want to have a unique .jar file and multiple [enties in the theme select input](https://github.com/keycloakify/keycloakify/assets/6702424/23ed36a4-e666-40e0-a89a-c74c4522e9cc), no it isn't possible unfortunately.  
Hi @mikalai-repin,

Thank you for reaching out! The answer depends on the specific scenario you have in mind. If you're looking to create a single .jar file with multiple entries in the [theme selection input](https://github.com/keycloakify/keycloakify/assets/6702424/23ed36a4-e666-40e0-a89a-c74c4522e9cc), unfortunately, that's not possible.

However, there are several strategies available for building polymorphic themes, as demonstrated in the following example:

https://github.com/keycloakify/keycloakify/assets/6702424/7f81fd5f-accc-4da7-8b83-7fa2c7ba6862

**Reply by mikalai-repin** [reply]
@garronej Thanks for the answer!)

**Comment by garronej** [comment]
Since 7.12 it's now possible: https://github.com/keycloakify/keycloakify/issues/354
